### ISSUE TITLE
installation issues 

### ISSUE THREAD

**Issue by salahaldin** [begriffs/postgrest#129]
Hello,

I am new to Haskell -help is appreciated- and I was not able to get postgrest up and running.  ubuntu the binaries raise this erorr.

./postgrest-0.2.5.0 
bash: ./postgrest-0.2.5.0: cannot execute binary file: Exec format error
System info
lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 14.10
Release:	14.10
Codename:	utopic
I have tried to compile it using 
cabal configure
cabal build
And I get this error 
Loading package entropy-0.3.5 ... linking ... done.
Loading package bcrypt-0.0.5 ... linking ... ghc: /home/salahaldin/.cabal/lib/i386-linux-ghc-7.6.3/bcrypt-0.0.5/libHSbcrypt-0.0.5.a: unknown symbol `_BF_body_r'
ghc: unable to load package `bcrypt-0.0.5'

Finally, could you please have a section on how to build the backages manually, also common issues. 
Regards

**Comment by begriffs** [comment]
Thanks for reporting this. I built the Linux binary on Heroku (cedar stack) which appears to be Ubuntu 10, so the binary is probably incompatible with modern Ubuntu.

Let's work together to figure out how to install it on your machine, and then I'll take what we learned and write an installation guide wiki.

My first guess is that you need SSL development libraries on your machine. Try this:

```bash
sudo apt-get install libssl-dev openssl
```

**Comment by salahaldin** [comment]
Thanks for support, libssl-dev and openssl are installed on my machine.

**Comment by begriffs** [comment]
So they were already installed and you still got that error?

**Comment by salahaldin** [comment]
Yep.

**Comment by begriffs** [comment]
OK, here is some more information to help us debug. I changed Circle CI to ghc 7.6.3 to make sure it wasn't a 7.6 problem and that built fine. Then user tommd on the haskell irc tried bcrypt on ubuntu and was able to build it:

> I ran  apt-get update && apt-get install haskell-platform && cabal update && cabal install bcrypt
> in a new ubuntu 14.04 (docker run -t -i ubuntu:14:04).  The install succeeded without issue.

Are you trying to build the project on your local machine? Does it have the haskell platform installed?

**Comment by TomMD** [comment]
I ran `apt-get update && apt-get install haskell-platform && cabal update && cabal install bcrypt`
in a new ubuntu 14.04 (`docker run -t -i ubuntu:14:04`).  The install succeeded without issue.  I assert this is either an architecture specific issue or the user has foobared their system somehow.

OTOH I notice `cabal install postgrest`, even after installing postgresql-server-dev-all, fails with:

```
src/Main.hs:7:8:
    Could not find module `Error'
    Perhaps you meant CError (needs flag -package haskell98-2.0.0.2)
```

This is with:

```
# ghc --version
The Glorious Glasgow Haskell Compilation System, version 7.6.3
# cabal --version
cabal-install version 1.16.0.2
using version 1.16.0 of the Cabal library 
```

So I believe a newer version of base is required.

**Comment by begriffs** [comment]
I wonder if this is a difference between cabal versions (I'm on 1.20) and how they treat the `Other-Modules` section of the cabal file. Can you try adding `Error` to the `Other-Modules` property of the `executable postgrest` section? If that works I'll update our cabal file.

**Comment by begriffs** [comment]
@salahaldin two more questions: do you have bcrypt installed (`apt-get install bcrypt`) and is your system 64-bit (`uname -a`)?

**Comment by TomMD** [comment]
@begriffs Indeed, any module used should be in the `Other-Modules` section afaik.  Adding `Error` in the .cabal fixed the issue I posted above.  With that adjustment I fully expect a fresh ubuntu 14.04 to successfully cabal install postrest via:

```
apt-get update
apt-get install postgresql postgresql-server-dev-all haskell-platform
cabal update
cabal install postgrest
```

**Comment by salahaldin** [comment]
bcrypt was not installed, so I have installed it but it did not help. Here are more information:
uname -a
Linux salaaldin 3.16.0-30-generic #40-Ubuntu SMP Mon Jan 12 22:07:27 UTC 2015 i686 i686 i686 GNU/Linux
ghc --version
The Glorious Glasgow Haskell Compilation System, version 7.6.3
cabal --version
cabal-install version 1.20.0.3
using version 1.20.0.2 of the Cabal library 

When I try to run the caommand cabal build to build from source code I get the above error 
 Loading package bcrypt-0.0.5 ... linking ... ghc: /root/.cabal/lib/i386-linux-ghc-7.6.3/bcrypt-0.0.5/libHSbcrypt-0.0.5.a: unknown symbol `_BF_body_r'
ghc: unable to load package `bcrypt-0.0.5'

using cabal install postgrest give the following 
src/Main.hs:7:8:
    Could not find module `Error'
    Perhaps you meant CError (needs flag -package haskell98-2.0.0.2)
    Use -v to see a list of the files searched for.
Failed to install postgrest-0.2.5.1
cabal: Error: some packages failed to install:
postgrest-0.2.5.1 failed during the building phase. The exception was:
ExitFailure 1

**Comment by salahaldin** [comment]
As indicated By  Thomas M. DuBuisson,  issue #6 might be the cause due to my system arch. So, I think you could close the issue.
Thanks for great support.

**Comment by begriffs** [comment]
@salahaldin can you try to build postgrest again? A new version of bcrypt was released which should fix the 32-bit build problem, and version 0.2.5.3 of postgrest requires the new bcrypt.

**Comment by salahaldin** [comment]
The istallation went fine. Thank you.
